radixdlt/babylon-wallet-android
# Pre-Auth validation

## Description
This PR validates the expiration time for a pre-auth request and informs the user if it is near to expiration or expired before showing the analysis.


## How to test
1. Connect to test dapp
2. Send a pre-auth with expiration after delay. This should show the analysis
3. Send a pre-auth with expiration at time with a timestamp in the future with more than 30 seconds difference. This should show the analysis
4. Send a pre-auth with expiration at time with a timestamp in the future with less or equal 30 seconds. This should show the error dialog to the user and dismiss the transaction. The appropriate error is sent to dapp. (`SUBINTENT_EXPIRATION_TOO_CLOSE`)
5. Send a pre-auth with expiration at time in the past. This should show the error dialog to the user and dismiss the transaction. The appropriate error is sent to dapp. (`EXPIRED_SUBINTENT`)
